src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -145,8 +147,10 @@ public AgentSettings LoadSettings()
                                                 args,
                                                 enforceSupplied);
                 Trace.Info("serverUrl: {0}", serverUrl);
+                Dictionary<string, string> credentialArgs = (args == null) ? new Dictionary<string, string>() : new Dictionary<string, string>(args);
```

Comment:
will use "var" as suggested. I was not sure, because there is an expression what to do.